### ISSUE TITLE
Live error in followup function with state/DQ channels

### ISSUE THREAD

**Issue by titodalcanton** [gwastro/pycbc#2567]
Another error arising from `is_extent_valid()`.
```
2019-03-11 13:50:32,711 node742 computing followup data for coinc
2019-03-11 13:50:32,712 node742 Generating SPAtmplt, 160.0s, 8430, starting from 25.2 Hz
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/mpi4py/__main__.py", line 7, in <module>
    main()
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/mpi4py/run.py", line 196, in main
    run_command_line(args)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/mpi4py/run.py", line 47, in run_command_line
    run_path(sys.argv[0], run_name='__main__')
  File "/usr/lib64/python2.7/runpy.py", line 240, in run_path
    return _run_module_code(code, init_globals, run_name, path_name)
  File "/usr/lib64/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/bin/pycbc_live", line 607, in <module>
    data_reader, bank)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/bin/pycbc_live", line 174, in check_coincs
    bank, followup_ifos)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/bin/pycbc_live", line 134, in compute_followup_data
    data_readers[ifo], bank, template_id, coinc_times)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/filter/matchedfilter.py", line 1768, in followup_event_significance
    if not data_reader.state.is_extent_valid(state_start_time, lookback):
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/frame/frame.py", line 745, in is_extent_valid
    data = self.raw_buffer[s:e]
  File "</home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/decorator.pyc:decorator-gen-120>", line 2, in __getitem__
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/types/array.py", line 67, in _convert
    return fn(self, *args)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/types/array.py", line 799, in __getitem__
    return self._getslice(index)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/types/timeseries.py", line 106, in _getslice
    ' not supported').format(index.start))
ValueError: Negative start index (-107424) not supported
```
Sounds like the lookback time is too large?

**Comment by titodalcanton** [comment]
Might be related to the issue fixed in #2575.

**Comment by titodalcanton** [comment]
Nope, it still happens:
```
2019-03-13 21:55:48,554 node1159 88: Filtering L1
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/mpi4py/__main__.py", line 7, in <module>
    main()
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/mpi4py/run.py", line 196, in main
    run_command_line(args)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/mpi4py/run.py", line 47, in run_command_line
    run_path(sys.argv[0], run_name='__main__')
  File "/usr/lib64/python2.7/runpy.py", line 240, in run_path
    return _run_module_code(code, init_globals, run_name, path_name)
  File "/usr/lib64/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/bin/pycbc_live", line 621, in <module>
    data_reader, bank)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/bin/pycbc_live", line 176, in check_coincs
    data_readers, bank, followup_ifos)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/bin/pycbc_live", line 134, in compute_followup_data
    data_readers[ifo], bank, template_id, coinc_times)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/filter/matchedfilter.py", line 1768, in followup_event_significance
    if not data_reader.state.is_extent_valid(state_start_time, lookback):
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/frame/frame.py", line 745, in is_extent_valid
    data = self.raw_buffer[s:e]
  File "</home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/decorator.pyc:decorator-gen-120>", line 2, in __getitem__
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/types/array.py", line 67, in _convert
    return fn(self, *args)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/types/array.py", line 799, in __getitem__
    return self._getslice(index)
  File "/home/pycbc.live/testing/O2_replay/pycbc_live_O2_replay_env/lib/python2.7/site-packages/pycbc/types/timeseries.py", line 106, in _getslice
    ' not supported').format(index.start))
ValueError: Negative start index (-107424) not supported
```
The start index is the same as the earlier error, maybe an off-by-one situation?

**Comment by ahnitz** [comment]
Probably not taking into account correctly when the buffer hasn't filled enough to be as long as the lookback time.

**Comment by ahnitz** [comment]
@titodalcanton I know how to further follow this one up on my end. Is someone in the LVC trying to track this one down?

**Comment by titodalcanton** [comment]
Liting and I are looking into it. Liting made some fake frames with state/DQ channels and we are trying to make the error reproducible offline. We can share the frames with you if we do not get too far with that.

Meanwhile I added a bit of verboseness and here is the outcome:
```
2019-03-26 15:07:15,319 node742 computing followup data for coinc
2019-03-26 15:07:15,565 node742 Generating SEOBNRv4_ROM, 40.0s, 237355, starting from 25.0 Hz
2019-03-26 15:07:15,667 node742 data_reader.state.raw_buffer time range is [1237672732,1237673244) sample rate 16
2019-03-26 15:07:15,667 node742 data_reader.dq.raw_buffer time range is [1237672732,1237673244) sample rate 16
2019-03-26 15:07:15,667 node742 state_start_time = 1237666043 bdur = 175
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/mpi4py/__main__.py", line 7, in <module>
    main()
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/mpi4py/run.py", line 196, in main
    run_command_line(args)
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/mpi4py/run.py", line 47, in run_command_line
    run_path(sys.argv[0], run_name='__main__')
  File "/usr/lib64/python2.7/runpy.py", line 240, in run_path
    return _run_module_code(code, init_globals, run_name, path_name)
  File "/usr/lib64/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/tito/projects/pycbc/live/multidetector/env/bin/pycbc_live", line 629, in <module>
    data_reader, bank)
  File "/home/tito/projects/pycbc/live/multidetector/env/bin/pycbc_live", line 181, in check_coincs
    data_readers, bank, followup_ifos)
  File "/home/tito/projects/pycbc/live/multidetector/env/bin/pycbc_live", line 134, in compute_followup_data
    data_readers[ifo], bank, template_id, coinc_times)
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/pycbc/filter/matchedfilter.py", line 1784, in followup_event_significance
    if not data_reader.state.is_extent_valid(state_start_time, bdur):
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/pycbc/frame/frame.py", line 745, in is_extent_valid
    data = self.raw_buffer[s:e]
  File "</home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/decorator.pyc:decorator-gen-120>", line 2, in __getitem__
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/pycbc/types/array.py", line 67, in _convert
    return fn(self, *args)
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/pycbc/types/array.py", line 799, in __getitem__
    return self._getslice(index)
  File "/home/tito/projects/pycbc/live/multidetector/env/lib/python2.7/site-packages/pycbc/types/timeseries.py", line 106, in _getslice
    ' not supported').format(index.start))
ValueError: Negative start index (-107024) not supported
```
Comparing the times (log times are PDT), the state and DQ channels appear to be nicely in sync with the analysis. But for some reason, `state_start_time` ends up being ~2 *hours* before the current time…

**Comment by titodalcanton** [comment]
Ah, isn't this simply the case that `data_reader.reduced_pad` is actually measured in samples, and is being mixed with quantities expressed in seconds?
```python
state_start_time = data_reader.strain.end_time \
                - data_reader.reduced_pad - bdur
```

**Comment by titodalcanton** [comment]
Fixed in #2577.
